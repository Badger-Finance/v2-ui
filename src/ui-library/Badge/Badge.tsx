import { Chip, makeStyles } from '@material-ui/core';
import clsx from 'clsx';
import React from 'react';

import { BadgeProps, BadgeType } from './Badge.types';

const useStyles = makeStyles({
	tag: {
		fontSize: '12px',
		alignItems: 'center',
		fontWeight: 700,
		height: 20,
	},
	newTag: {
		background: '#FFFFFF1A',
		color: '#FFFFFF',
		border: '0.5px solid #FFFFFF',
	},
	guardedTag: {
		background: '#FBEE781A',
		color: '#FBEE78',
		border: '0.5px solid #FBEE78',
	},
	obsoleteTag: {
		backgroundColor: '#FF614D1A',
		color: '#FF614D',
		border: '0.5px solid #FF614D',
	},
	experimentalTag: {
		backgroundColor: '#FFFFFF1A',
		color: '#FFFFFF99',
		border: '0.5px solid #FFFFFF99',
	},
	executedTag: {
		backgroundColor: '#29CCBB1A',
		color: '#29CCBB',
		border: '0.5px solid #29CCBB',
	},
});

export function Badge({ type }: BadgeProps): JSX.Element | null {
	const classes = useStyles();

	switch (type) {
		case BadgeType.FEATURED:
			return <Chip label="Featured" className={clsx(classes.tag, classes.newTag)} size="small" />;
		case BadgeType.GUARDED:
			return <Chip className={clsx(classes.tag, classes.guardedTag)} size="small" label="Guarded" />;
		case BadgeType.DISCONTINUED:
			return <Chip className={clsx(classes.tag, classes.obsoleteTag)} size="small" label="Discontinued" />;
		case BadgeType.EXECUTED:
			return <Chip className={clsx(classes.tag, classes.executedTag)} size="small" label="Executed" />;
		case BadgeType.EXPERIMENTAL:
			return <Chip className={clsx(classes.tag, classes.experimentalTag)} size="small" label="Experimental" />;
		default:
			return null;
	}
}
