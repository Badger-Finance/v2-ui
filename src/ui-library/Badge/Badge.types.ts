export enum BadgeType {
  FEATURED = 'featured',
  GUARDED = 'guarded',
  DISCONTINUED = 'discontinued',
  EXECUTED = 'executed',
  EXPERIMENTAL = 'experimental',
}

export interface BadgeProps {
  type: BadgeType;
}
