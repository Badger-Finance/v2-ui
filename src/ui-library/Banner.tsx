import { Grid, Link, makeStyles, Typography } from '@material-ui/core';
import React from 'react';

export interface BannerProps {
  message: string;
  linkText?: string;
  link?: string;
  closeElement?: React.ReactNode;
}

const useStyles = makeStyles({
  root: {
    minHeight: 48,
    padding: '14px 26px',
    background: '#BDE0FF',
  },
  messageLinkContainer: {
    display: 'flex',
  },
  linkIcon: {
    marginRight: 6,
  },
  actionButton: {
    color: '#2E44C0',
  },
  actionLabel: {
    fontSize: 14,
    fontWeight: 500,
    letterSpacing: 1.25,
    textTransform: 'uppercase',
  },
  link: {
    color: '#2E44C0',
    letterSpacing: '0.0025em',
    fontSize: 14,
    fontWeight: 'bold',
  },
  message: {
    fontWeight: 'normal',
    fontSize: 14,
    letterSpacing: 0.25,
    color: 'rgba(0, 0, 0, 0.87)',
    marginRight: 12,
  },
  actionContainer: {
    paddingLeft: 40,
  },
  closeIcon: {
    color: '#2E44C0',
    margin: '-12px',
    '& svg': {
      fontSize: 24,
    },
  },
});

const Banner = ({ message, link, linkText, closeElement = null }: BannerProps): JSX.Element => {
  const classes = useStyles();
  return (
    <Grid container alignItems="center" className={classes.root} justifyContent="space-between">
      <Grid item container xs alignItems="center" className={classes.messageLinkContainer}>
        <Grid item>
          <Typography className={classes.message}>{message}</Typography>
        </Grid>
        {link && linkText && (
          <Grid item>
            <Link href={link} rel="noreferrer" target="_blank" className={classes.link}>
              <img className={classes.linkIcon} src="/assets/icons/banner-link-icon.svg" alt="banner link icon" />
              {linkText}
            </Link>
          </Grid>
        )}
      </Grid>
      {!!closeElement && (
        <Grid item xs="auto" className={classes.actionContainer}>
          {closeElement}
        </Grid>
      )}
    </Grid>
  );
};

export default Banner;
