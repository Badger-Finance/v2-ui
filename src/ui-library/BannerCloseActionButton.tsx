import { Button, makeStyles } from '@material-ui/core';
import React from 'react';

const useStyles = makeStyles({
  actionButton: {
    color: '#2E44C0',
  },
  actionLabel: {
    fontSize: 14,
    fontWeight: 500,
    letterSpacing: 1.25,
    textTransform: 'uppercase',
  },
});

interface Props {
  text: string;
  onClose: () => void;
}

const BannerCloseActionButton = ({ text, onClose }: Props): JSX.Element => {
  const classes = useStyles();
  return (
    <Button
      variant="text"
      className={classes.actionButton}
      classes={{ label: classes.actionLabel }}
      onClick={onClose}
    >
      {text}
    </Button>
  );
};

export default BannerCloseActionButton;
