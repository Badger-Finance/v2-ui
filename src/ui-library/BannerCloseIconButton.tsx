import React from 'react';
import CloseIcon from '@material-ui/icons/Close';
import { IconButton, makeStyles } from '@material-ui/core';

const useStyles = makeStyles({
	closeIcon: {
		color: '#2E44C0',
		margin: '-12px',
		'& svg': {
			fontSize: 24,
		},
	},
});

interface Props {
	onClose: () => void;
}

const BannerCloseIconButton = ({ onClose }: Props): JSX.Element => {
	const classes = useStyles();
	return (
		<IconButton onClick={onClose} className={classes.closeIcon} size="medium" aria-label="close banner">
			<CloseIcon />
		</IconButton>
	);
};

export default BannerCloseIconButton;
