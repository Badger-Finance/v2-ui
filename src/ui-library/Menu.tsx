import React from 'react';
import { List, ListProps, makeStyles } from '@material-ui/core';
import clsx from 'clsx';

const useStyles = makeStyles({
	root: {
		background: '#3a3a3a',
		color: 'rgba(255, 255, 255, 0.87)',
		fontSize: 16,
		fontWeight: 400,
	},
});

const Menu = (props: ListProps): JSX.Element => {
	const classes = useStyles();
	return (
		<List
			component="nav"
			{...props}
			classes={{
				...(props.classes ?? {}),
				root: clsx(classes.root, props.classes?.root),
			}}
		>
			{props.children}
		</List>
	);
};

export default Menu;
