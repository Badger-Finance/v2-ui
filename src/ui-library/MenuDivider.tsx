import { Divider, DividerProps, makeStyles } from '@material-ui/core';
import clsx from 'clsx';
import React from 'react';

const useStyles = makeStyles({
  root: {
    backgroundColor: 'rgba(255, 255, 255, 0.3)',
  },
});

const MenuDivider = (props: DividerProps): JSX.Element => {
  const classes = useStyles();
  return (
    <Divider
      {...props}
      classes={{
        ...(props.classes ?? {}),
        root: clsx(classes.root, props.classes?.root),
      }}
    >
      {props.children}
    </Divider>
  );
};

export default MenuDivider;
