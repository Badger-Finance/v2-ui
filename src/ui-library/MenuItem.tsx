import React from 'react';
import { ListItem, ListItemProps, makeStyles } from '@material-ui/core';
import clsx from 'clsx';

const useStyles = makeStyles({
	button: {
		'&:hover': {
			background: '#545454',
		},
		'&:disabled': {
			color: 'rgba(255, 255, 255, 0.15)',
		},
	},
});

//https://github.com/mui/material-ui/issues/14971#issuecomment-616791594
interface CustomMenuItemProps extends ListItemProps {
	button?: any;
}

const MenuItem = (props: CustomMenuItemProps): JSX.Element => {
	const classes = useStyles();
	return (
		<ListItem
			{...props}
			classes={{
				...(props.classes ?? {}),
				button: clsx(classes.button, props.classes?.button),
			}}
		>
			{props.children}
		</ListItem>
	);
};

export default MenuItem;
