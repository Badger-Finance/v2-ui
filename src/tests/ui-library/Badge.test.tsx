import React from 'react';
import { Badge, BadgeType } from '../../ui-library/Badge';
import { customRender } from '../Utils';

const testCases = Object.values(BadgeType).map((type) => [type]);

describe('Badge', () => {
	test.each([...testCases])('renders for type %s', (type) => {
		const { container } = customRender(<Badge type={type} />);
		expect(container).toMatchSnapshot();
	});
});
