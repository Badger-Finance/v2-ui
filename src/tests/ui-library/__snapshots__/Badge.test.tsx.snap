// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders for type discontinued 1`] = `
<div>
  <div
    class="MuiChip-root-7 makeStyles-tag-1 makeStyles-obsoleteTag-4 MuiChip-sizeSmall-8"
  >
    <span
      class="MuiChip-label-29 MuiChip-labelSmall-30"
    >
      Discontinued
    </span>
  </div>
</div>
`;

exports[`Badge renders for type executed 1`] = `
<div>
  <div
    class="MuiChip-root-7 makeStyles-tag-1 makeStyles-executedTag-6 MuiChip-sizeSmall-8"
  >
    <span
      class="MuiChip-label-29 MuiChip-labelSmall-30"
    >
      Executed
    </span>
  </div>
</div>
`;

exports[`Badge renders for type experimental 1`] = `
<div>
  <div
    class="MuiChip-root-7 makeStyles-tag-1 makeStyles-experimentalTag-5 MuiChip-sizeSmall-8"
  >
    <span
      class="MuiChip-label-29 MuiChip-labelSmall-30"
    >
      Experimental
    </span>
  </div>
</div>
`;

exports[`Badge renders for type featured 1`] = `
<div>
  <div
    class="MuiChip-root-7 makeStyles-tag-1 makeStyles-newTag-2 MuiChip-sizeSmall-8"
  >
    <span
      class="MuiChip-label-29 MuiChip-labelSmall-30"
    >
      Featured
    </span>
  </div>
</div>
`;

exports[`Badge renders for type guarded 1`] = `
<div>
  <div
    class="MuiChip-root-7 makeStyles-tag-1 makeStyles-guardedTag-3 MuiChip-sizeSmall-8"
  >
    <span
      class="MuiChip-label-29 MuiChip-labelSmall-30"
    >
      Guarded
    </span>
  </div>
</div>
`;
